### ISSUE TITLE
Separate email field in member profile/registration system and email template updates

### ISSUE THREAD

**Issue by shirey** [hubmapconsortium/member-ui#92]

**Comment by yuanzhou** [comment]
9/19/2024

Deb requested to add a simple formatting tweak to the email template to keep the ordering consistent: 

It would be helpful if the specifics for each HM resource were listed in the same order for both old and new requests, and the resource choice values were also in the same order.  For example, for an profile update the requests are listed:
Old requests
1 HuBMAP Data Via Globus
2 Collaboration Portal
3 [protocols.io](http://protocols.io/)
4 HuBMAP Google Drive Share
5 HuBMAP GitHub Repository
6 HuBMAP Slack Workspace
And new requests are in a different order
Collaboration portal
Globus
Google Drive
GitHub
Slack
[protocols.io](http://protocols.io/)
and the values (if you will ) are listed in this order:
GLobus username
Drive email
GitHub username
Slack email
[protocols.io](http://protocols.io/) email

**Comment by humadragonbane** [comment]
performed migration of existing Production WP files to replace the Development WP installation.

backups of dev + prod made

```
# DEV
[root@DEV opt]# ./backup-wp.sh
[root@DEV opt]# mv hubmap-wordpress-2024-10-10-14-14-15.tar.gz /home/cdb135/
[root@DEV opt]# mv hubmap-wordpress-dev-db-2024-10-10-14-14-03.sql.gz /home/cdb135/

[root@DEV opt]# tar -cvf wordpressbackup-dev.tar.gz /opt/hubmap/
[root@DEV opt]# systemctl stop nginx

##created new venv
$ sudo python3 -m venv /opt/hubmap/member-ui/venv-member-ui
$ sudo chown ec2-user:ec2-user venv-member-ui/ -R
(venv-member-ui) $ pip install -r requirements.txt

## Had to update the requirements.txt using the prod requirements file to include uWSGI==2.0.26

# Migration Changes for tables
$./wp search-replace 'https://hubmapconsortium.org' 'https://dev.hubmapconsortium.org' --all-tables --path=/opt/hubmap/wp-site/
Success: Made 26 replacements.

# restarted nginx + php + custom uwsgi config
# Copied wp-config.php from old dev to new dev
# Verified that both sites are up and functional (Visual exam + sign in attempt through profile.dev)


#### File transfer notes
## SCP
C:\Users\CDB135\.ssh>scp -i hubmappem ec2-user@PROD:/tmp/wordpress-prod-hubmap.tar.gz C:\Users\CDB135\.ssh
C:\Users\CDB135\.ssh>scp -i hubmappem C:\Users\CDB135\.ssh\wordpress-prod-hubmap.tar.gz ec2-user@DEV:/tmp/wordpress-prod-hubmap.tar.gz 

C:\Users\CDB135\.ssh>scp -i hubmappem ec2-user@PROD:/tmp/hubmap-wordpress-2024-10-10-18-15-20.tar.gz C:\Users\CDB135\.ssh
C:\Users\CDB135\.ssh>scp -i hubmappem C:\Users\CDB135\.ssh\hubmap-wordpress-2024-10-10-18-15-20.tar.gz ec2-user@DEV:/tmp/hubmap-wordpress-2024-10-10-18-15-20.tar.gz

C:\Users\CDB135\.ssh>scp -i hubmappem ec2-user@PROD:/tmp/hubmap-wordpress-db-2024-10-10-18-15-12.sql.gz C:\Users\CDB135\.ssh
C:\Users\CDB135\.ssh>scp -i hubmappem C:\Users\CDB135\.ssh\hubmap-wordpress-db-2024-10-10-18-15-12.sql.gz ec2-user@DEV:/tmp/hubmap-wordpress-db-2024-10-10-18-15-12.sql.gz 
```

**Comment by BirdMachine** [comment]
@debnigra   Cheers! Just some questions on the email templates when you have a moment. 
What's meant by "Old Requests" and "New Requests"? Is the desire to rearrange the new request emails to match the old request format? Or are there 'Old' requests still coming in?  

Old requests
1 HuBMAP Data Via Globus
2 Collaboration Portal
3 protocols.io
4 HuBMAP Google Drive Share
5 HuBMAP GitHub Repository
6 HuBMAP Slack Workspace

- What do you mean by "HuBMAP Data Via Globus"? Is this a collection of values associated with Globus Identity? (IE Name First Name Last Name Email and such)?
- What do you mean by "Collaboration Portal"? What Fields did that cover?


And new requests are in a different order
Collaboration portal
Globus
Google Drive
GitHub
Slack
protocols.io

- "Collaboration Portal" I'm still in the dark on this one; what fields does this cover? 
- Is this "Globus" the Same Content as the above "HuBMAP Data Via Globus"? 
- I'm assuming "HuBMAP Google Drive Share" is the same as "Google Drive" here
- GitHub: it looks like we ask for the user info now, not a Repository link. Same regarding Slack / Slack Workspace. Are the Github and Slack user fields new, but we're still expecting to gave a repository and workspace link, too? Or have they been replaced with the user name Values? Do we need to include both now to Align Old and New Requests? 

And the values (if you will ) are listed in this order:
GLobus username
Drive email
GitHub username
Slack email
protocols.io email

- Listed Where? Are these the only fields you're getting in this e mail? are "The Values" where, exactly? In the Email? In another page? 
- It looks like the "New" Request order lines up with the order of the Values, except for the missing"Collaboration Portal" from the request. Does this mean nothing is being captured for this value set? 

It might be better rather than to sort out what linked to what in what order, we make a list for what you want the final order to be going forward (Which looks pretty consistent aside from swapping Globus / Collaboration and moving protocols.io to the end.)

**Comment by debnigra** [comment]
@BirdMachine 

Old/New requests refer to info from the original registration request vs. info from a Profile Update request.

Yes, HuBMAP Data via Globus/Globus  and HuBMAP Google Drive Share/Google Drive are the same fields.

"HuBMAP Data via Globus"  just means membership in the HuBMAP Globus READ group. Only  the member's Globus identity is collected here.

Membership in the READ is necessary but not sufficient to be able to upload data to the data portal. For upload access, you also have to be a member of your group's specific Globus group.  Those permissions are requested through a ticket but do not involve the registration system.  

"Collaboration portal"  means access to the member portal on the website. This is granted by the user being created on WordPress when we approve the registration. 

For GitHub and Slack, we only collect username and email, respectively. Members are added to all repos in GitHub by default.

Both the registration the profile update emails include all values.  Notably the profile update email includes both the original (Old) and updated (New) values. 

PLEASE I would love to have the fields and the values in a consistent order. You choose, I don't really care  - plese let's have Old/New "resources", if you will (GLobus, Slack, github, etc) in the same order and the values also in the same order.  

Thanks!

**Comment by yuanzhou** [comment]
Addressed via PR #93
